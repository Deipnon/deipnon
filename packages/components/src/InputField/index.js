// @flow
import * as React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/core'

import theme from '@deipnon/themes/lib'

type PropsType = {|
	hasError: boolean,
	onChange: (e: SyntheticEvent<HTMLInputElement>) => void,
	placeholder: ?string,
	type: string,
	value: ?string
|}

const inputBaseStyle = css`
	display: block;
	padding: ${theme.styles.formPaddings};
	font-size: ${theme.styles.formFontSize};
	font-weight: ${theme.styles.formFontWeight};
	line-height: ${theme.styles.formLineHeight};
	color: ${theme.styles.formColor};
	background-color: ${theme.styles.formBgColor};
	background-clip: padding-box;
	border: ${theme.styles.formBorderWidth} solid ${theme.styles.formBorderColor};
	width: 100%;
`

const withError = ({ hasError }: PropsType) => {
	if (hasError) {
		return css`
			border-color: ${theme.styles.formErrorBorderColor};
		`
	}
}

const InputElement = styled('input')(
	inputBaseStyle,
	withError
)

const InputField = (props: PropsType) => <InputElement {...props} />

InputField.defaultProps = {
	type: 'text',
	hasError: false
}

export default InputField
