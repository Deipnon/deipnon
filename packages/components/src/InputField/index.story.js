import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, boolean, text } from '@storybook/addon-knobs/react'

import InputField from '.'

const stories = storiesOf('Input Field', module)
stories.addDecorator(withKnobs)

stories.add('default', () => (

	<InputField
		hasError={boolean('Has Error', false)}
		placeholder={text('Placeholder', '')}
	/>

))
