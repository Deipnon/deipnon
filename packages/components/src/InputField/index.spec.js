import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-emotion'

import InputField from '.'

describe('InputField', () => {
	test('renders correctly', () => {
		const tree = renderer.create(<InputField placeholder='test' />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
