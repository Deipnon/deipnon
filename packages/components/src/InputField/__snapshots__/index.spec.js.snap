// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<input
  className="css-tjurys"
  placeholder="test"
  type="text"
/>
`;
